pkg/core/interop/context.go

```diff
@@ -334,6 +334,15 @@ func (ic *Context) SpawnVM() *vm.VM {
 	return v
 }
 
+// ReuseVM resets given VM and allows to reuse it in the current context.
+func (ic *Context) ReuseVM(v *vm.VM) {
+	v.Reset(ic.Trigger)
+	v.LoadToken = ic.LoadToken
+	v.SyscallHandler = ic.SyscallHandler
+	v.SetPriceGetter(ic.GetPrice)
+	ic.VM = v
```

Comment:
We also need to set `GasLimit` to `-1` here. it is needed for cases where `GasLimit` was modified by the previous call to match the previous behavior. We have such situation on `PostPersist` running if there were some transactions in block.

Reply:
I thought we could do without it, but you're probably right, even though it works this way.

Reply:
Fixed.